tasks/eva_2380/find_FP_variants.py

```diff
@@ -0,0 +1,161 @@
+                    for tag in ['AS', 'XS', 'XN', 'XM', 'XG', 'NM', 'MD', 'BD', 'BT']
+                ]),
+                file=open_csv)
+
+
+def get_alignment_records(variant_ids, bam_file):
+    print('Read alignment BAM: ' + bam_file, file=sys.stderr)
+    variant_id_to_bam_records = defaultdict(list)
+    with pysam.AlignmentFile(bam_file, "rb") as bam_in:
+        for sam_record in bam_in:
+            sp_name = sam_record.query_name.split('|')
+            rs_id = sp_name[4]
+            if rs_id in variant_ids:
+                variant_id_to_bam_records[rs_id].append(sam_record)
+    return variant_id_to_bam_records
+
+
+def process_files(realigned_vcf_file, assessment_vcf_file, source_vcf, truth_vcf, bam_files, old_ref, new_ref):
+    assessment_result_per_variants = assessment_result_per_variants_from_vcf(assessment_vcf_file)
+    variant_id_to_assessment_result, variant_id_to_alignment_result = get_variant_id(assessment_result_per_variants, realigned_vcf_file)
+    variant_id_to_source_record = get_variant_from_vcf(variant_id_to_assessment_result, source_vcf)
+    variant_id_to_truth_record = get_variant_from_vcf(variant_id_to_assessment_result, truth_vcf)
+    alignment_record_lists = []
+    for bam_file in bam_files:
+        alignment_record_lists.append(get_alignment_records(variant_id_to_assessment_result, bam_file))
+
+    flank_len = 100
+    old_fasta = pysam.FastaFile(old_ref)
+    new_fasta = pysam.FastaFile(new_ref)
+
+    for variant_id in variant_id_to_assessment_result:
+        source_var = variant_id_to_source_record[variant_id]
+        truth_var = variant_id_to_truth_record[variant_id]
+
+        print('################################')
+        print(variant_id)
+        print('In source:' + str(source_var))
+        print('In truth:' + str(truth_var))
+        print('In realignment:' + str(variant_id_to_alignment_result[variant_id]))
+        for alignment_records in alignment_record_lists:
+            for alignment_record in alignment_records[variant_id]:
+                print('Alignments:' + alignment_record.to_string())
+            print('----')
+
+        old_ref_left = fetch_bases(old_fasta, source_var.chrom, source_var.pos - flank_len, flank_len)
+        old_ref = fetch_bases(old_fasta, source_var.chrom, source_var.pos, len(source_var.ref))
+        old_ref_right = fetch_bases(old_fasta, source_var.chrom, source_var.pos + len(source_var.ref), flank_len)
+
+        new_ref_left = fetch_bases(new_fasta, truth_var.chrom, truth_var.pos - flank_len, flank_len)
+        new_ref = fetch_bases(new_fasta, truth_var.chrom, truth_var.pos, len(truth_var.ref))
+        new_ref_right = fetch_bases(new_fasta, truth_var.chrom, truth_var.pos + len(truth_var.ref), flank_len)
+
+        print(old_ref_left + "*" + old_ref + "*" + old_ref_right)
+        print(new_ref_left + "*" + new_ref + "*" + new_ref_right)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
```

Comment:
i would at least put a description string, out of pity for any future EVA team members